### PR TITLE
net-libs/shairplay: New package needed by media-tv/kodi

### PR DESCRIPTION
**Pull request by frace**
Gentoo-bug: 538932, 468558, 458734

"Apple airplay and raop protocol server" allows Kodi to receive streams from Apple devices via Airplay.

### CODE REVIEW

File: net-libs/shairplay/shairplay-0_pre20160304.ebuild [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+src_configure() {
+	econf \
+		$(use_enable static-libs static)
+}
+
+src_install() {
+	default
+	prune_libtool_files --all
```
Comment (anthonyryan1) [inline net-libs/shairplay/shairplay-0_pre20160304.ebuild]: Replace this with:
```
find "${D}" -name '*.la' -delete || die
```

For rationale, see: https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Handling_Libtool_Archives

File: net-libs/shairplay/shairplay-0_pre20160304.ebuild [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI="6"
+
+inherit autotools eutils
```
Comment (anthonyryan1) [inline net-libs/shairplay/shairplay-0_pre20160304.ebuild]: eutils is only used for prune_libtool_files, and can be removed once that is changed.

**Comment by anthonyryan1** [comment]
Also really happy to see this PR, this has been an outstanding issue for years now.

**Comment by frace** [comment]
@anthonyryan1 Thanks for your input.

> Also really happy to see this PR, this has been an outstanding issue for years now.
Yes, it's been nearly four years now since I reported the issue initially. ^^

File: net-libs/shairplay/shairplay-0_pre20160304.ebuild (hunk 2 above)
Comment (frace) [inline net-libs/shairplay/shairplay-0_pre20160304.ebuild]: Fixed in 8f8e129289ddfff2f3eb6bd29704b93f9face2b4

File: net-libs/shairplay/shairplay-0_pre20160304.ebuild (hunk 1 above)
Comment (frace) [inline net-libs/shairplay/shairplay-0_pre20160304.ebuild]: Fixed in 8f8e129289ddfff2f3eb6bd29704b93f9face2b4

**Comment by candrews** [comment]
I think this PR should just cover the new package. Modifications to the Kodi ebuild(s) should be handled separately.

**Comment by frace** [comment]
@candrews Thanks for your input - that's correct. I adjusted the PR accordingly.

**Comment by candrews** [comment]
A few things:
* You only express a runtime dependency on openssl and libao. Based on what https://github.com/juhovh/shairplay says, I think that's not quite right. I think you need:
```
DEPENDS="
>=dev-libs/openssl-1.0.0
media-libs/libao
dev-libs/libltdl
net-dns/avahi[mdnsresponder-compat]
"
RDEPENDS="${DEPENDS}"
```
* metadata.xml isn't right. I think you should have proxy maintainers mentioned as you're not a Gentoo developer, take a look at https://github.com/gentoo/gentoo/blob/0de592f0c63eeca0756d1950372cfa2fac7e85ee/media-plugins/kodi-audiodecoder-modplug/metadata.xml for an example
* Does this work with other SSL implementations than OpenSSL? If so, update the dependency accordingly and add appropriate USE flag(s), see https://github.com/gentoo/gentoo/blob/44f733e8bf8069c0898d99df9b764f472d95bd79/media-tv/kodi/kodi-9999.ebuild for example
* You need to squash commits and only have one commit, and the commit message should be formatted in the standard way (ex, it should be "Gentoo-bug: XXX", not "Gentoo bug: XXX"), see https://wiki.gentoo.org/wiki/Gentoo_git_workflow#Commit_message_format

**Comment by frace** [comment]
> You only express a runtime dependency on openssl and libao. Based on what https://github.com/juhovh/shairplay says, I think that's not quite right. I think you need:

Looks like you are right. I'll fix that.
Tbh I didn't compare the README for quite a long time since it worked quite well as it is. I even can't remember what the initial requirements were when I wrote the first version of the ebuild. :)

> metadata.xml isn't right. I think you should have proxy maintainers mentioned as you're not a Gentoo developer, take a look at https://github.com/gentoo/gentoo/blob/0de592f0c63eeca0756d1950372cfa2fac7e85ee/media-plugins/kodi-audiodecoder-modplug/metadata.xml for an example

That's true.
But I'm not quite sure if I fit into the role of a reliable proxy maintainer due to time constraints. I'll have to think about it. Should I rather omit `metadata.xml` for the time being?

> Does this work with other SSL implementations than OpenSSL? If so, update the dependency accordingly and add appropriate USE flag(s), see https://github.com/gentoo/gentoo/blob/44f733e8bf8069c0898d99df9b764f472d95bd79/media-tv/kodi/kodi-9999.ebuild for example

I'll check and fix it if necessary.

> You need to squash commits and only have one commit, and the commit message should be formatted in the standard way (ex, it should be "Gentoo-bug: XXX", not "Gentoo bug: XXX"), see https://wiki.gentoo.org/wiki/Gentoo_git_workflow#Commit_message_format

I see.

File: net-libs/shairplay/shairplay-0_pre20160304.ebuild [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+DESCRIPTION="Apple airplay and raop protocol server"
+HOMEPAGE="https://github.com/juhovh/shairplay"
+
+LICENSE="BSD LGPL-2.1 MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="static-libs"
+
+RDEPEND="
+	>=dev-libs/openssl-1.0.0
```
Comment (SoapGentoo) [inline net-libs/shairplay/shairplay-0_pre20160304.ebuild]: needs `:0=`, if you have time/commitment, you can also try whether it works with libressl, but don't feel obligated

File: net-libs/shairplay/shairplay-0_pre20160304.ebuild [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+inherit autotools 
+
+if [[ ${PV} == "9999" ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="git://github.com/juhovh/shairplay.git"
+else
+	EGIT_COMMIT="498bc5bcdd305e04721f94a04b9f26a7da72673f"
+	SRC_URI="https://github.com/juhovh/shairplay/archive/${EGIT_COMMIT}.tar.gz -> ${P}.tar.gz"
+	S="${WORKDIR}/${PN}-${EGIT_COMMIT}"
```
Comment (SoapGentoo) [inline net-libs/shairplay/shairplay-0_pre20160304.ebuild]: if you like, you can also `inherit vcs-snapshot`, which sets this itself

**Comment by candrews** [comment]
> metadata.xml isn't right. I think you should have proxy maintainers mentioned as you're not a Gentoo developer
> 
> That's true.
> But I'm not quite sure if I fit into the role of a reliable proxy maintainer due to time constraints. I'll have to think about it. Should I rather omit metadata.xml for the time being?

You can have yourself be the maintainer - a block like this:
```
    <maintainer type="person">
        <email>YOUR EMAIL ADDRESS</email>
        <name>YOUR NAME</name>
    </maintainer>
    <maintainer type="project">
        <email>proxy-maint@gentoo.org</email>
        <name>Proxy Maintainers</name>
</maintainer>
```

**Comment by frace** [comment]
> You can have yourself be the maintainer

Yes, but as I wrote, I'm not quite sure if I fit into the role of a reliable proxy maintainer due to time constraints. The question at this point is which route to go then - omit `metadata.xml` entirely or just leave it as is without a specific block containing data pointing to a specific maintainer?

**Comment by candrews** [comment]
@SoapGentoo surely has an opinion here, but IMHO, as long as you have the element with `proxy-maint@gentoo.org` then any proxy maintainer can update the package and things will be okay.

**Comment by anthonyryan1** [comment]
If we need someone to list as a proxy maintainer, I'm willing to do it for this package. While I haven't used this in 6+ months I can test it as needed and begin using it again.

**Comment by frace** [comment]
Quick notes in regards to dependencies:


dev-libs/libltdl
Pulled in by sys-devel/libtool.

>=dev-libs/openssl-1.0.0
Seems not to be required at all. At least I couldn't find any hints inside the README. Although the sources have a pointer:

`# This library depends on 3rd party libraries
libshairplay_la_LIBADD = crypto/libcrypto.la alac/libalac.la
libshairplay_la_LDFLAGS = -no-undefined -version-info 0:0:0`

I tested it with and without openssl installed and couldn't tell any difference. Opinions?

media-libs/libao
Only required if one wants to build a binary `/usr/bin/shairplay`.

net-dns/avahi[mdnsresponder-compat]
Only needed if someone wants to use the binary `/usr/bin/shairplay` directly.

**Comment by candrews** [comment]
If `dev-libs/openssl` isn't a dependency, I see no reason to depend on it :)

**Comment by candrews** [comment]
You could add a USE flag to conditionally build the `shairplay` binary and only then depend on `media-libs/libao` and `net-dns/avahi[mdnsresponder-compat]`

**Comment by frace** [comment]
> You could add a USE flag to conditionally build the shairplay binary and only then depend on media-libs/libao and net-dns/avahi[mdnsresponder-compat]

That's what I've been ending up while researching deps. :)
I'm not quite sure about the reference inside the sources to libcrypto (openssl) and libalac (alac_decoder) though.

File: net-libs/shairplay/shairplay-0_pre20160304.ebuild (hunk 3 above)
Comment (frace) [inline net-libs/shairplay/shairplay-0_pre20160304.ebuild]: Fixed in 621719dcc4ed53b4e63e63c4aa7f5c6ebc4a2505
I left an issue @juhovh to clear our questions in regards to ssl.

File: net-libs/shairplay/shairplay-0_pre20160304.ebuild (hunk 4 above)
Comment (frace) [inline net-libs/shairplay/shairplay-0_pre20160304.ebuild]: Fixed in 621719dcc4ed53b4e63e63c4aa7f5c6ebc4a2505

**Comment by frace** [comment]
> as long as you have the element with proxy-maint@gentoo.org then any proxy maintainer can update the package and things will be okay.

Fixed in 621719dcc4ed53b4e63e63c4aa7f5c6ebc4a2505

File: net-libs/shairplay/shairplay-0_pre20160304.ebuild (hunk 4 above)
Comment (frace) [inline net-libs/shairplay/shairplay-0_pre20160304.ebuild]: ...I think it's fine as it is for now. I tested it against media-tv/kodi with and without openssl installed - works in both cases.

**Comment by frace** [comment]
Squashed commits with 9381f4fabbebf11fea5c5552d5db92901cbc59b5

File: net-libs/shairplay/shairplay-0_pre20170118.ebuild [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+KEYWORDS="~amd64 ~x86"
+IUSE="alac static-libs tools"
+
+DEPEND="
+	sys-devel/libtool
+	tools? ( media-libs/libao )
+"
+
+RDEPEND="
+	alac? (
```
Comment (anthonyryan1) [inline net-libs/shairplay/shairplay-0_pre20170118.ebuild]: There's a trailing space on this line.

Comment (frace) [inline net-libs/shairplay/shairplay-0_pre20170118.ebuild]: Fixed with d7862bfff0fd9c62df452fbc6786523aa3541000

**Comment by frace** [comment]
Some notes:
I tested this ebuild with every version of `>=media-tv/kodi-16.1`. Currently it seems like Airplay does only work with the v16 branch. Yet I didn't dive deep into the issue and couldn't find a reason why it doesn't work as expected. Only events that are obvious are:
1. `>=media-tv/kodi-17` use Cmake as a build system
2. Recent changes in [`AirPlayServer.cpp`](https://github.com/xbmc/xbmc/commit/76d9cd2d495b7eac2a3fa937bb0f8240da94e7da)

**Comment by candrews** [comment]
@frace looks like you intended to include a -9999 ebuild as a copy of shairplay-0_pre20170118.ebuild - can you please add that?

File: net-libs/shairplay/metadata.xml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+    <longdescription>Apple airplay and raop protocol server</longdescription>
+    <use>
+        <flag name='alac'>Add support for Apple Lossless Audio Codec files</flag>
+        <flag name='static-libs'>Build static versions of dynamic libraries as well</flag>
+        <flag name='tools'>Build and install a binary shairplay in /usr/bin</flag>
+    </use>
+    <maintainer type="project">
```
Comment (SoapGentoo) [inline net-libs/shairplay/metadata.xml]: missing a primary person, I suggest @candrews

File: net-libs/shairplay/shairplay-0_pre20170118.ebuild [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+DESCRIPTION="Apple airplay and raop protocol server"
+HOMEPAGE="https://github.com/juhovh/shairplay"
+
+LICENSE="BSD LGPL-2.1 MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="alac static-libs tools"
+
+DEPEND="
+	sys-devel/libtool
```
Comment (SoapGentoo) [inline net-libs/shairplay/shairplay-0_pre20170118.ebuild]: `libtool` shouldn't be necessary if you're inheriting `autotools` anyway.

File: net-libs/shairplay/metadata.xml (hunk 6 above)
Comment (frace) [inline net-libs/shairplay/metadata.xml]: Fixed with 3d528af8834d346458a46c7d6f61846e0ecb8417

File: net-libs/shairplay/shairplay-0_pre20170118.ebuild (hunk 7 above)
Comment (frace) [inline net-libs/shairplay/shairplay-0_pre20170118.ebuild]: Fixed with 3d528af8834d346458a46c7d6f61846e0ecb8417

**Comment by frace** [comment]
@candrews 
>  looks like you intended to include a -9999 ebuild as a copy of shairplay-0_pre20170118.ebuild - can you please add that?

Fixed with 3d528af8834d346458a46c7d6f61846e0ecb8417

**Comment by candrews** [comment]
Seems good to me... What do you say, @SoapGentoo ?

File: net-libs/shairplay/shairplay-0_pre20170118.ebuild [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+	EGIT_COMMIT="498bc5bcdd305e04721f94a04b9f26a7da72673f"
+	SRC_URI="https://github.com/juhovh/shairplay/archive/${EGIT_COMMIT}.tar.gz -> ${P}.tar.gz"
+fi
+
+DESCRIPTION="Apple airplay and raop protocol server"
+HOMEPAGE="https://github.com/juhovh/shairplay"
+LICENSE="BSD LGPL-2.1 MIT"
+
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
```
Comment (anthonyryan1) [inline net-libs/shairplay/shairplay-0_pre20170118.ebuild]: As QA just helpfully pointed out, to use this, KEYWORDS needs to be moved into the else branch of the if 9999 conditional.

File: net-libs/shairplay/shairplay-9999.ebuild [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+	EGIT_COMMIT="498bc5bcdd305e04721f94a04b9f26a7da72673f"
+	SRC_URI="https://github.com/juhovh/shairplay/archive/${EGIT_COMMIT}.tar.gz -> ${P}.tar.gz"
+fi
+
+DESCRIPTION="Apple airplay and raop protocol server"
+HOMEPAGE="https://github.com/juhovh/shairplay"
+LICENSE="BSD LGPL-2.1 MIT"
+
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
```
Comment (anthonyryan1) [inline net-libs/shairplay/shairplay-9999.ebuild]: Ditto here.

**Comment by candrews** [comment]
@frace always run repoman before pushing :-)

**Comment by frace** [comment]
@candrews Correct! Looks like I'm a *little* bit out of practice. ^^

File: net-libs/shairplay/shairplay-0_pre20170118.ebuild (hunk 8 above)
Comment (frace) [inline net-libs/shairplay/shairplay-0_pre20170118.ebuild]: Fixed with 572e036c9cc45db418a9182efbf7c5c9b3d55b3c

File: net-libs/shairplay/shairplay-9999.ebuild (hunk 9 above)
Comment (frace) [inline net-libs/shairplay/shairplay-9999.ebuild]: Fixed with 572e036c9cc45db418a9182efbf7c5c9b3d55b3c

**Comment by frace** [comment]
Thanks for your support and accepting the pull request!

**Pull request closed without merging**